closure/templates/soy_plugin.bzl

```diff
@@ -0,0 +1,44 @@
+# Copyright 2019 The Closure Rules Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+SoyPluginInfo = provider()
+
+def _impl(ctx):
+    files = []
+    classpath = []
+    for dep in ctx.attr.deps:
+        for jar in dep[JavaInfo].transitive_runtime_deps.to_list():
+            classpath.append(jar.path)
+            files.append(jar)
+    return [SoyPluginInfo(
+        module = ctx.attr.module,
+        deps = ctx.attr.deps,
+        classpath = depset(direct=classpath),
+        files = files,
+    )]
+
+soy_plugin = rule(
+    implementation = _impl,
+    attrs = {
+        "module": attr.string(
+            doc = "fully-qualified class name of extension module",
+            mandatory = True,
+        ),
+        "deps": attr.label_list(
```

Comment:
I can't think of any scenario where this wouldn't be a single java_library, so I briefly considered using an `attr.label` instead of `label_list`, but I figured using the common deps attribute was worth it just for consistency with other rule types.

Reply:
sgtm